src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -635,8 +635,8 @@ private static long DateToTicks(int year, int month, int day)
                 ThrowHelper.ThrowArgumentOutOfRange_BadYearMonthDay();
             }
 
-            int y = year - 1;
-            int n = y * 365 + y / 4 - y / 100 + y / 400 + days[month - 1] + day - 1;
+            uint y = (uint)(year - 1);
+            uint n = y * 365 + y / 4 - y / 100 + y / 400 + (uint)days[month - 1] + (uint)day - 1;
             return n * TicksPerDay;
         }
```

Comment:
y is `int` but always is in the [1..9999] range so can be treated as unsigned (`X /u C` is faster than `X / C`)